Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByObject)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The dataset object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        public Dataset Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatasetId for the datset reference object. Must be unique within the project, and must be 1024 
+        /// characters or less. It must only contain letters (a-z, A-Z), numbers (0-9), or underscores (_).
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByValues)]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "friendly name" field for the dataset. Used for a descriptive name for the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "description" field for the dataset. Used for a user-friendly description of the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The default lifetime for tables in the dataset to exist in ms. Minimum is 3600000 (1hr).  
+        /// When the expiration time is reached, the table will be automatically deleted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(3600000, long.MaxValue)]
+        public long Timeout { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            Dataset newData;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    newData = Dataset;
+                    break;
+                case ParameterSetNames.ByValues:
+                    if (DatasetId == null)
```

Comment:
Instead of this, you can put [ValidateNotNullOrEmpty] attribute on top of DatasetId so PowerShell will automatically throw an error if user supplies a null or empty DatasetId.